code/core/src/builder-manager/index.ts

```diff
@@ -200,6 +200,7 @@ const starter: StarterFunction = async function* starterGeneratorFn({
 
   // Build additional global values
   const globals: Record<string, any> = await buildFrameworkGlobalsFromOptions(options);
+  globals.STORYBOOK_NETWORK_ADDRESS = options.networkAddress;
```

Comment:
The `buildFrameworkGlobalsFromOptions`-function above is the place to modify/create the `globals` object based on what's in `options`.

Don't mutate the object afterwards.